react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,77 @@
 import React from 'react';
 
+import {calculateWinner} from './components/utils'
 import styles from './styles.module.scss';
 import Board from './components/Board';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends React.Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  handleClick = (i) => {
+    const { history } = this.state;
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{
+        squares: squares,
+      }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    });
+  }
+
+  jumpTo(step) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: (step % 2) === 0,
+    });
+  }
+
+  render() {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
```

Comment:
You are calculating the winner in each render. You should be calculating this in each click, and save the component state to prevent calculating it again.
When you do a click, you could save in the winner in the state, and use it from there in the render.
In handleClick, you would change the `calculateWinner` condition for just checking if there is a winner in this.state.winner. That way, you would only be calculating the winner once per click.